spec/outputs/s3_spec.rb

```diff
@@ -258,4 +258,25 @@
     end
   end
 
+  context "plugin close" do
+    let(:temporary_directory) { Stud::Temporary.pathname }
+    let(:options) { super().merge({ 'temporary_directory' => temporary_directory, 'rotation_strategy' => 'time', 'time_file' => 0.1 }) }
+
+    before do
+      allow(subject).to receive(:bucket_resource).and_return(mock_bucket)
+    end
+
+    it "removes empty directory" do
+      subject.register
+      expect(Dir.exist?(temporary_directory)).to be_truthy
+
+      subject.multi_receive_encoded(events_and_encoded) # creates a temporary dir
+      expect((Dir.entries(temporary_directory) - %w{ . .. }).empty?).to be_falsey
+
+      sleep(20) # let file rotation happen and leave new empty dir
```

Comment:
```suggestion
      subject.instance_variable_get(:@periodic_check).execute
```
I'm not sure if this suggestion is a Ruby's bad practice, but I think we could force the rotate here instead of waiting 20 secs. Other option would be extracting the @periodic_check body code into a method, and just invoke it :)

Reply:
Yeah, from the best practise I also usually avoid accessing internal properties but for our case I do believe we save resource cost and we have a single isolated method with periodic checker, will update with required changes.